server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -155,52 +154,7 @@ public final class IndexModule {
         Property.NodeScope
     );
 
-    /** Which lucene file extensions to load with the mmap directory when using hybridfs store. This settings is ignored if {@link #INDEX_STORE_HYBRID_NIO_EXTENSIONS} is set.
-     *  This is an expert setting.
-     *  @see <a href="https://lucene.apache.org/core/9_5_0/core/org/apache/lucene/codecs/lucene95/package-summary.html#file-names">Lucene File Extensions</a>.
-     *
-     * @deprecated This setting will be removed in OpenSearch 3.x. Use {@link #INDEX_STORE_HYBRID_NIO_EXTENSIONS} instead.
-     */
-    @Deprecated
-    public static final Setting<List<String>> INDEX_STORE_HYBRID_MMAP_EXTENSIONS = Setting.listSetting(
```

Comment:
@andrross @dblock @nknize seems like we are really rushing the removal here: the feature will be deprecated in 2.10.0 but already removed in 3.0.0, it looks to be very short notice ...

Reply:
I think semver policy technically allows for deprecating something in any minor version and removing in the next major, even if those end up being consecutive releases. Is that wrong?

For a widely used feature that requires complicated changes changes to move away from then for sure we'd want to give more runway than that. For an expert setting like this, I'm more comfortable with the faster removal schedule. We can carry this for another major version though if you think this will be too disruptive.

Reply:
> I think semver policy technically allows for deprecating something in any minor version and removing in the next major, even if those end up being consecutive releases. Is that wrong?

Not generally wrong, doing that in `2.0` would be 100% OK, in `2.10.0` raises the question how much time (till `3.0.0`) we give users to migrate

Reply:
I am inclined to make this change in `3.0`. It is a major release so users can expect these type of changes.

Also even though we deprecate the setting in `2.10`, this does not guarantee users switch from using `mmap.extensions` to `nio.extensions`. Many will ignore the warning so keeping a longer migration window makes no difference.

Reply:
@andrross have no objections to speed it up if you think so

> Many will ignore the warning so keeping a longer migration window makes no difference.

We don't know how widely this setting is being used, may be no one uses it, giving sufficient is the only thing we could do.